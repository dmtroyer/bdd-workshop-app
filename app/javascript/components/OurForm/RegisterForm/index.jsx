import React from 'react';
import { Form } from 'reactstrap';
import PropTypes from 'prop-types';
import ActionButton from '../ActionButton';
import StepFormGroup from '../StepFormGroup';

const RegisterForm = props => (
  <Form onSubmit={props.handleSubmit}>
    {(props.step === 'handle' || !props.step) && (
      <StepFormGroup
        {...props}
        step="handle"
        label="Handle"
        placeholder="input a custom handle"
      />
    )}
    {(props.step === 'email' || !props.step) && (
      <StepFormGroup
        {...props}
        step="email"
        label="Email"
        type="email"
        placeholder="input your email"
      />
    )}
    <ActionButton>{props.step ? 'Next' : 'Submit'}</ActionButton>
  </Form>
);
RegisterForm.propTypes = {
  handleSubmit: PropTypes.func,
  step: PropTypes.string,
};

export default RegisterForm;
