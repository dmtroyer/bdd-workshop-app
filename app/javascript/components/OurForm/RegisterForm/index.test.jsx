import React from 'react';
import { shallow } from 'enzyme';
import RegisterForm from '.';

describe('RegisterForm', () => {
  it('Renders a register form', () => {
    const wrapper = shallow(<RegisterForm />);
    expect(wrapper).toMatchInlineSnapshot(`
<Form
  tag="form"
>
  <StepFormGroup
    label="Handle"
    placeholder="input a custom handle"
    step="handle"
  />
  <StepFormGroup
    label="Email"
    placeholder="input your email"
    step="email"
    type="email"
  />
  <ActionButton>
    Submit
  </ActionButton>
</Form>
`);
  });

  it('Renders an Action button with text submit', () => {
    const wrapper = shallow(<RegisterForm />);
    expect(wrapper.find('ActionButton').prop('children')).toEqual('Submit');
  });

  it('Renders the handle with label and placeholder', () => {
    const wrapper = shallow(<RegisterForm step="handle" />);
    expect(wrapper.find('StepFormGroup')).toMatchInlineSnapshot(`
<StepFormGroup
  label="Handle"
  placeholder="input a custom handle"
  step="handle"
/>
`);
  });

  it('Renders the email with label and placeholder', () => {
    const wrapper = shallow(<RegisterForm step="email" />);
    expect(wrapper.find('StepFormGroup')).toMatchInlineSnapshot(`
<StepFormGroup
  label="Email"
  placeholder="input your email"
  step="email"
  type="email"
/>
`);
  });

  it('Submit calls handleSubmti', () => {
    const mockHandleSubmit = jest.fn();
    const wrapper = shallow(<RegisterForm handleSubmit={mockHandleSubmit} />);
    wrapper.simulate('submit');
    expect(mockHandleSubmit).toHaveBeenCalledTimes(1);
  });
});
