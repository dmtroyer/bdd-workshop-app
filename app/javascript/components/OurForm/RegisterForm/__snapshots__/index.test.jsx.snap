// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm Renders a register form 1`] = `
<Form
  tag="form"
>
  <ActionButton>
    Next
  </ActionButton>
</Form>
`;
